util.go

```diff
@@ -22,13 +22,25 @@ func fileExists(fname string) bool {
 	if err != nil {
 		return false
 	}
+	// file exists but might not be readable
+	f, err := os.Open(fname)
+	if err != nil {
+		log.Error("Cannot open file : [%v]", err)
+		os.Exit(1)
+	}
+	f.Close()
 	return true
 }
 
 func readConfig(fname string) DNSConfig {
 	var conf DNSConfig
 	// Practically never errors
-	_, _ = toml.DecodeFile(fname, &conf)
+	// but will if there is an error in the config file
+	_, err := toml.DecodeFile(fname, &conf)
+	if err != nil {
+		log.Error("Error decoding configuration file [%v]", err)
```

Comment:
You should probably use `log.Errorf` here

Reply:
I made the requested changes and submitted them (see below).

Reply:
This `log.Error` should be changed to `log.Errorf` still.